express-app/app.controllers.js

```diff
@@ -1,31 +1,29 @@
-const express = require('express');
-const app = express();
 const endpointsJson = require('../endpoints.json');
-const { fetchTopics } = require('./app.models');
+const { fetchTopics, fetchArticleID } = require('./app.models');
 
 exports.healthCheck = (req, res) => {
   res.status(200).send({ endpoints: endpointsJson });
 };
 
 exports.getTopics = (req, res, next) => {
-  const { slug, description } = req.query;
-  if (slug && !['mitch', 'cats', 'paper'].includes(slug)) {
-    return next({ status: 404, msg: 'Not found' });
-  }
-  if (
-    description &&
-    ![
-      'The man, the Mitch, the legend',
-      'Not dogs',
-      'what books are made of',
-    ].includes(description)
-  ) {
-    return next({ status: 404, msg: 'Not found' });
-  }
   fetchTopics()
-    .then((result) => {
-      console.log({ status: 200, msg: `200: Server Running Okay` });
-      res.status(200).send({ result });
+    .then((topic) => {
+      res.status(200).send({ topic });
+    })
+    .catch(next);
+};
+
+exports.getArticleID = (req, res, next) => {
+  const { article_id } = req.params;
+  if (!article_id || !Number(article_id)) {
+    return next({ status: 400, msg: 'Bad Request' });
+  }
+  fetchArticleID(article_id)
+    .then((article) => {
+      if (!article) {
```

Comment:
We should handle errors (i.e. invoke next) within our .catch()s - otherwise there would be no reason to have them! So consider how we could get our model to let the controller know there's an error, and allow it to respond appropriately :)